pkgs/dart_tooling_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -201,6 +211,89 @@ base mixin DartToolingDaemonSupport on ToolsSupport {
     );
   }
 
+  /// Retrieves runtime errors from the currently running app.
+  ///
+  /// If more than one debug session is active, then it just uses the first one.
+  ///
+  // TODO: support passing a debug session id when there is more than one debug
+  // session.
+  Future<CallToolResult> runtimeErrors(CallToolRequest request) async {
+    return _callOnVmService(
+      callback: (vmService) async {
+        final errors = <String>[];
+        try {
+          await vmService.streamListen(EventStreams.kExtension);
+          await vmService.streamListen(EventStreams.kStderr);
+
+          // We need to listen to streams with history so that we can get errors
+          // that occurred before this tool call.
+          // TODO(https://github.com/dart-lang/ai/issues/57): this can result in
+          // duplicate errors that we need to de-duplicate somehow.
+          final extensionEvents = vmService.onExtensionEventWithHistory.listen((
+            Event e,
+          ) {
+            if (e.extensionKind == 'Flutter.Error') {
+              // TODO(https://github.com/dart-lang/ai/issues/57): consider
+              // pruning this content down to only what is useful for the LLM to
+              // understand the error and its source.
+              errors.add(e.json.toString());
+            }
+          });
+          final stderrEvents = vmService.onStderrEventWithHistory.listen((
+            Event e,
+          ) {
+            final message = decodeBase64(e.bytes!);
+            // TODO(https://github.com/dart-lang/ai/issues/57): consider
+            // pruning this content down to only what is useful for the LLM to
+            // understand the error and its source.
+            errors.add(message);
+          });
+
+          // Await a short delay to allow the error events to come over the open
+          // Streams.
+          await Future<void>.delayed(const Duration(seconds: 1));
+
+          await vmService.streamCancel(EventStreams.kExtension);
+          await vmService.streamCancel(EventStreams.kStderr);
+          await extensionEvents.cancel();
+          await stderrEvents.cancel();
+
+          if (errors.isEmpty) {
+            return CallToolResult(
+              content: [TextContent(text: 'No runtime errors found.')],
+            );
+          }
+          return CallToolResult(
+            content: [
+              TextContent(
+                text:
+                    'Found ${errors.length} '
+                    'error${errors.length == 1 ? '' : 's'}:\n',
+              ),
+              ...errors.map((e) => TextContent(text: e.toString())),
+            ],
+          );
+        } on RPCError catch (e) {
```

Comment:
Fwiw, see https://modelcontextprotocol.io/specification/2025-03-26/server/tools#error-handling, it might be technically correct to just let these errors be reported as normal RPC errors (ie: "Server Error"s in this case?).

Reply:
removed the special handling for RPCErrors and just handled all exceptions in the catch block.